### PR TITLE
feat: [CDS-78030]: Use the individual template schema from the parsed individual schemas to serve the UI/BE requests and added ttl for parsed schemas

### PR DESCRIPTION
**Pull request by utkarshchoubey-harness**
Use the individual template schema from the parsed individual schemas to serve the UI requests

Use individual template Schema from the parsed individual schemas to serve the BE template save/update 

Added 15mins ttl for individual schema parser for preqa env

Tests Done - 
* Fetch Chaos step Schema
* Fetch Complete Template Schema
* Fetch CI Stage Schema
* When fetching complete template schema we will fetch from schema repo for PREQA env else from Resource
* Added support for Versioning for future (v0 is default)

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)

### CODE REVIEW

**Comment by utkarshchoubey-harness** [comment]
trigger messagecheck

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Comment by utkarshchoubey-harness** [comment]
trigger ut7

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`template-service/contracts/openapi/v1/openapi.yaml` line 513 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA5k5T-eD_N0-Eo-bH:-NeA5k5T-eD_N0-Eo-bI:bxawfjz) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/contracts/openapi/v1/openapi.yaml#L513)):*
> ```YAML
>       requestBody:
>         $ref: '#/components/requestBodies/TemplateImportRequestBody'
>   /v1/template-schema/individual-schema:
> ```

shouldn't this be internal endpoint?

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`template-service/contracts/openapi/v1/openapi.yaml` line 515 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA5mio7nmfUYSdR5XL:-NeA5mio7nmfUYSdR5XM:b-e8v0p6) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/contracts/openapi/v1/openapi.yaml#L515)):*
> ```YAML
>   /v1/template-schema/individual-schema:
>     get:
>       summary: Your GET endpoint
> ```

If its public, then summary should be fixed here

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`template-service/service/src/main/java/io/harness/template/resources/SchemasApiImpl.java` line 28 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA5uVNBBil4UcCP9GC:-NeA5uVNBBil4UcCP9GD:btp5yrf) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/service/src/main/java/io/harness/template/resources/SchemasApiImpl.java#L28)):*
> ```Java
> @NextGenManagerAuth
> public class SchemasApiImpl implements SchemasApi {
>   private final NGTemplateSchemaService ngTemplateSchemaService;
> ```

add line spacing

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java` line 181 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA67wfC6G34O8miG8q:-NeA67wfC6G34O8miG8r:b70f4rl) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java#L181)):*
> ```Java
> 
>   @Override
>   public ObjectNode getStaticSchemaForAllEntities(String nodeGroup, String nodeType, String version) {
> ```

shouldn't the method name be : "getStaticSchema"?

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java` line 65 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA6Eie0JNIE6w4WLlN:-NeA6Eie0JNIE6w4WLlO:bm49yjn) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java#L65)):*
> ```Java
>   @Inject TemplateSchemaFetcher templateSchemaFetcher;
> 
>   @Inject TemplateSchemaParserV0 templateSchemaParserV0;
> ```

why are we injecting direct parsers here? It should come via some factory or helper

**Review by utkarshchoubey-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 9 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`template-service/contracts/openapi/v1/openapi.yaml` line 513 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA5k5T-eD_N0-Eo-bH:-NeCC5dCC2oeaRXc_FwS:b352wts) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/contracts/openapi/v1/openapi.yaml#L513)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

shouldn't this be internal endpoint?
</blockquote></details>

Done. yes it should be internal
___
*[`template-service/contracts/openapi/v1/openapi.yaml` line 515 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA5mio7nmfUYSdR5XL:-NeCC7S47-9AQ7oMNGPx:bxxru7r) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/contracts/openapi/v1/openapi.yaml#L515)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

If its public, then summary should be fixed here
</blockquote></details>

Done. NA as it it internal
___
*[`template-service/service/src/main/java/io/harness/template/resources/SchemasApiImpl.java` line 28 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA5uVNBBil4UcCP9GC:-NeCCBo49EoEPEw3Ucnr:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/service/src/main/java/io/harness/template/resources/SchemasApiImpl.java#L28)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

add line spacing
</blockquote></details>

Done.
___
*[`template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java` line 65 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA6Eie0JNIE6w4WLlN:-NeCCWpa5JPD3vVrSE3S:b-3rxnts) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java#L65)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

why are we injecting direct parsers here? It should come via some factory or helper
</blockquote></details>

Done.
We removed the factory in previous prs, as we moved individual parsers to diff services to fetch schema
Is there any better way to do it ?
___
*[`template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java` line 181 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA67wfC6G34O8miG8q:-NeCCHDVBi_Y6909snCT:b-2r36vt) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java#L181)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

shouldn't the method name be : "getStaticSchema"?
</blockquote></details>

Done. It implies we are using individual static schema, getStaticSchema there is already a func with that name, checking we can improve the name

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r1, 4 of 8 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 9 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java` line 182 at r3](https://reviewable.io/reviews/harness/harness-core/52463#-NeCXKow6NXJzCeJluQG:-NeCXKow6NXJzCeJluQH:b8cciue) ([raw file](https://github.com/harness/harness-core/blob/f24fecfd936aefced20aad2a31fd06905d6a9232/pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java#L182)):*
> ```Java
>        * TODO - @UtkarshChoubey make this time as configurable number
>        */
>       return System.currentTimeMillis() - lastInitializedTimeForPipelineV0Parser >= 900000;
> ```

can you move into another method , todo configurable is not required



_Code quote:_
```Java
System.currentTimeMillis() - lastInitializedTimeForPipelineV0Parser >= 900000;
```
___
*[`template-service/service/src/main/java/io/harness/pms/yaml/individualschema/TemplateSchemaParserV0.java` line 142 at r3](https://reviewable.io/reviews/harness/harness-core/52463#-NeCY-vg2T_nfLpNx-49:-NeCY-vg2T_nfLpNx-4A:bidul04) ([raw file](https://github.com/harness/harness-core/blob/f24fecfd936aefced20aad2a31fd06905d6a9232/template-service/service/src/main/java/io/harness/pms/yaml/individualschema/TemplateSchemaParserV0.java#L142)):*
> ```Java
>        */
>       return System.currentTimeMillis() - lastInitializedTimeForTemplateV0Parser >= 900000;
>     }
> ```

can this method be common for both pipeline and template? and other entities if required can use it



_Code quote:_
```Java
    if (PRE_QA.equals(env)) {
      /*
       * We will reinitialise the individual schema in 15 min for stress env
       * TODO - @UtkarshChoubey make this time as configurable number
       */
      return System.currentTimeMillis() - lastInitializedTimeForTemplateV0Parser >= 900000;
    }
```

File: 953-yaml-commons/src/main/java/io/harness/pms/yaml/individualschema/AbstractStaticSchemaParser.java [hunk 1]
```diff
@@ -307,4 +306,13 @@ String getTypeFromObjectNode(ObjectNode objectNode) {
     }
     return null;
   }
+
+  Boolean checkIfPreQaEnvAndLastInitializedTimeGreaterThanThresholdTime(long lastInitializedTime) {
```
Comment (utkarshchoubey-harness) [inline 953-yaml-commons/src/main/java/io/harness/pms/yaml/individualschema/AbstractStaticSchemaParser.java:310]: make name generic

**Review by utkarshchoubey-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 9 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java` line 65 at r2](https://reviewable.io/reviews/harness/harness-core/52463#-NeA6Eie0JNIE6w4WLlN:-NeD4LyL-CwuuwCtqpWY:b-qkvxzh) ([raw file](https://github.com/harness/harness-core/blob/25a3134ec0e3d329068ee9cd00018479555d503f/template-service/service/src/main/java/io/harness/template/services/NGTemplateSchemaServiceImpl.java#L65)):*
<details><summary><i>Previously, utkarshchoubey-harness wrote…</i></summary><blockquote>

Done.
We removed the factory in previous prs, as we moved individual parsers to diff services to fetch schema
Is there any better way to do it ?
</blockquote></details>

https://harness.atlassian.net/browse/CDS-79221

**Review by archit-harness — DISMISSED** [review]
> Reviewed 1 of 8 files at r3, 2 of 2 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 9 file(s) to review. 1 change requests. 5. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java` line 182 at r3](https://reviewable.io/reviews/harness/harness-core/52463#-NeCXKow6NXJzCeJluQG:-NeD50tADPDbs0shrkh0:boeo9lc) ([raw file](https://github.com/harness/harness-core/blob/f24fecfd936aefced20aad2a31fd06905d6a9232/pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java#L182)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can you move into another method , todo configurable is not required
</blockquote></details>

done, discussed to be done in another PR
___
*[`template-service/service/src/main/java/io/harness/pms/yaml/individualschema/TemplateSchemaParserV0.java` line 142 at r3](https://reviewable.io/reviews/harness/harness-core/52463#-NeCY-vg2T_nfLpNx-49:-NeD54PW9U8Pgf766GEo:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/f24fecfd936aefced20aad2a31fd06905d6a9232/template-service/service/src/main/java/io/harness/pms/yaml/individualschema/TemplateSchemaParserV0.java#L142)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can this method be common for both pipeline and template? and other entities if required can use it
</blockquote></details>

done

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Comment by utkarshchoubey-harness** [comment]
trigger ut4,ut7

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 9 file(s) to review. 1 change requests. 5. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java` line 40 at r6](https://reviewable.io/reviews/harness/harness-core/52463#-NeDiZENA4dKBcVBF3-k:-NeDiZENA4dKBcVBF3-l:b-dy4vqn) ([raw file](https://github.com/harness/harness-core/blob/562960ffd07c801e209ec926afc9ae8c6d1f00d1/pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java#L40)):*
> ```Java
>   @Inject SchemaFetcher schemaFetcher;
> 
>   static long lastInitializedTimeForPipelineV0Parser;
> ```

this variable name should come from abstract class and should be generic

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 9 file(s) to review. 1 change requests. 5. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java` line 53 at r6](https://reviewable.io/reviews/harness/harness-core/52463#-NeDid6l1FMxnr9bozqd:-NeDid6l1FMxnr9bozqe:btvqfh3) ([raw file](https://github.com/harness/harness-core/blob/562960ffd07c801e209ec926afc9ae8c6d1f00d1/pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java#L53)):*
> ```Java
> 
>   @Override
>   void initParser() {
> ```

This should be default implementation in abstract class. Only if someone wants to override it, they should be able to do so.

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 9 file(s) to review. 1 change requests. 5. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java` line 172 at r6](https://reviewable.io/reviews/harness/harness-core/52463#-NeDik9O2tYODcjsrcvz:-NeDik9O2tYODcjsrcw-:b-d2mwls) ([raw file](https://github.com/harness/harness-core/blob/562960ffd07c801e209ec926afc9ae8c6d1f00d1/pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java#L172)):*
> ```Java
> 
>   @Override
>   boolean shouldInitParser() {
> ```

this should also be abstracted out to top layer.

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by utkarshchoubey-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 9 file(s) to review. 1 change requests. 9. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`953-yaml-commons/src/main/java/io/harness/pms/yaml/individualschema/AbstractStaticSchemaParser.java` line 314 at r6](https://reviewable.io/reviews/harness/harness-core/52463#-NeDiIqRA7xf2D6WYbRk:-NeDulm23Ket_NNkz6A4:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/562960ffd07c801e209ec926afc9ae8c6d1f00d1/953-yaml-commons/src/main/java/io/harness/pms/yaml/individualschema/AbstractStaticSchemaParser.java#L314)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

time should be a constant here, not hardcoded
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java` line 40 at r6](https://reviewable.io/reviews/harness/harness-core/52463#-NeDiZENA4dKBcVBF3-k:-NeDuorx2X4mrO3P0hMZ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/562960ffd07c801e209ec926afc9ae8c6d1f00d1/pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java#L40)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

this variable name should come from abstract class and should be generic
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java` line 53 at r6](https://reviewable.io/reviews/harness/harness-core/52463#-NeDid6l1FMxnr9bozqd:-NeDuptA0f8jbtR3AbAE:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/562960ffd07c801e209ec926afc9ae8c6d1f00d1/pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java#L53)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

This should be default implementation in abstract class. Only if someone wants to override it, they should be able to do so.
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java` line 172 at r6](https://reviewable.io/reviews/harness/harness-core/52463#-NeDik9O2tYODcjsrcvz:-NeDusVD8PzqY4bAbWK8:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/562960ffd07c801e209ec926afc9ae8c6d1f00d1/pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserV0.java#L172)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

this should also be abstracted out to top layer.
</blockquote></details>

Done.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 6 files at r1, 3 of 8 files at r3, 2 of 2 files at r4, 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52463)* status: :shipit: complete! ✓

___
*[`953-yaml-commons/src/main/java/io/harness/pms/yaml/individualschema/AbstractStaticSchemaParser.java` line 43 at r7](https://reviewable.io/reviews/harness/harness-core/52463#-NeGgCkNBiGVZAC44YKr:-NeGgCkNBiGVZAC44YKs:bquvtnz) ([raw file](https://github.com/harness/harness-core/blob/a5e44c912c557a09e7c0f547f065ea717e3221b7/953-yaml-commons/src/main/java/io/harness/pms/yaml/individualschema/AbstractStaticSchemaParser.java#L43)):*
> ```Java
>   protected String PRE_QA = "stress";
> 
>   final long MAX_TIME_TO_REINITIALIZE_PARSER = 900000;
> ```

Lets use Durations.toMillis(TimeUnit.Minutes, 15) rather than this big number

File: template-service/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -510,12 +510,54 @@ paths:
         - $ref: '#/components/parameters/AccountHeaderParam'
       requestBody:
         $ref: '#/components/requestBodies/TemplateImportRequestBody'
+  /v1/template-schema/individual-schema:
+    get:
+      summary: Your GET endpoint
```
Comment (mankrit-singh) [inline template-service/contracts/openapi/v1/openapi.yaml:515]: please add a summary

**Review by mankrit-singh — APPROVED** [review]

File: template-service/contracts/openapi/v1/openapi.yaml (hunk 2 above)
Comment (mankrit-singh) [reply template-service/contracts/openapi/v1/openapi.yaml:515]: feel free to take this up in the next PR

**Comment by utkarshchoubey-harness** [comment]
trigger review

**Pull request merged**